quote/quoteclient.go

```diff
@@ -1,89 +1,39 @@
 package quoteclient
 
 import (
-	"bufio"
+	"github.com/shopspring/decimal"
+	"os"
+	"net/http"
+	"log"
 	"fmt"
-	"net"
-	"regexp"
-	"seng468/transaction-server/logger"
 	"strconv"
-	"time"
-
-	"github.com/patrickmn/go-cache"
-	"github.com/shopspring/decimal"
+	"io/ioutil"
 )
 
-type QuoteClientI interface {
-	Query(string, string, int) (decimal.Decimal, error)
-}
-
-type QuoteClient struct {
-	name   string
-	addr   string
-	cache  *cache.Cache
-	re     *regexp.Regexp
-	logger logger.Logger
-}
-
-type QuoteReply struct {
-	quote decimal.Decimal
-	stock string
-	user  string
-	time  uint64
-	key   string
-}
-
-func NewQuoteClient(logger logger.Logger) *QuoteClient {
-	re := regexp.MustCompile("(?P<quote>.+),(?P<stock>.+),(?P<user>.+),(?P<time>.+),(?P<key>.+)")
-	return &QuoteClient{
-		name:   "quoteserve",
-		addr:   "quoteserve.seng:4444",
-		cache:  cache.New(time.Minute, time.Minute),
-		re:     re,
-		logger: logger,
-	}
-}
+var addr = os.Getenv("quoteaddr")
+var port = os.Getenv("quoteport")
 
-func (q *QuoteClient) Query(u string, s string, transNum int) (decimal.Decimal, error) {
-	quote, found := q.cache.Get(s)
-	if found {
-		d, _ := decimal.NewFromString(quote.(string))
-		return d, nil
+func Query(user string, stock string, transNum int) (decimal.Decimal, error) {
+	req, err := http.NewRequest("GET", addr + ":" + port + "/quote", nil)
+	if err != nil {
+		log.Print(err)
```

Comment:
We should probably return something here, it looks like it will crash if it tries to continue after this on failure.